### PR TITLE
Delete unused `engine_hash.dart` script (and test), simplify `engine_hash.sh`.

### LINKED ISSUE #160527: Update `engine_hash.{dart|sh}`
https://github.com/flutter/flutter/blob/master/dev/tools/bin/engine_hash.dart
https://github.com/flutter/flutter/blob/master/dev/tools/bin/engine_hash.sh

need to be updated to be the same as 
https://github.com/flutter/flutter/blob/2e78e749e62f73de5a343836916a7b6d5550d51f/bin/internal/update_dart_sdk.sh

> (its basically the git hash); that will fix google3 bot when we get there.

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/flutter/flutter/issues/160527.

We don't use the Dart script anyway, and the shell script could be simplified to the single use in g3 (cl/688973229).

### CODE REVIEW

File: dev/tools/bin/engine_hash.sh [hunk 1]
```diff
@@ -5,70 +5,27 @@
...
+# See cl/688973229.
 #
 # -------------------------------------------------------------------------- #
 
-# TODO(codefu): Add a test that this always outputs the same hash as
-# `engine_hash.dart` when the repositories are merged
-
-STRATEGY=head
+# Test for fusion repository
+if [ -f "$FLUTTER_ROOT/DEPS" ] && [ -f "$FLUTTER_ROOT/engine/src/.gn" ]; then
```
Comment (jtmcdole) [inline dev/tools/bin/engine_hash.sh:17]: Is FLUTTER_ROOT set already?

Comment (matanlurey) [reply dev/tools/bin/engine_hash.sh]: Simplified further, and fixed, and tested :)

File: dev/tools/bin/engine_hash.sh [hunk 2]
```diff
@@ -5,70 +5,65 @@
...
+    GIT_BIN="$GIT"
+fi
 
-LSTREE=$(git ls-tree -r $BASE engine DEPS)
-if [ ${#LSTREE} -eq 0 ]; then
-    echo >&2 Error calculating engine hash: Not in a monorepo
-    exit 1
+# Test for fusion repository
+if [ -f "$FLUTTER_ROOT/DEPS" ]; then
+    ENGINE_VERSION=$($GIT_BIN -C "$FLUTTER_ROOT" merge-base HEAD upstream/master)
```
Comment (jtmcdole) [inline dev/tools/bin/engine_hash.sh:61]: This needs to be origin/master if the google3 script is running from a fresh checkout.

**Review by jtmcdole — APPROVED** [review]
One nit

File: dev/tools/bin/engine_hash.sh (hunk 2 above)
Comment (matanlurey) [reply dev/tools/bin/engine_hash.sh]: Done.

**Pull request merged**